components/mediation-initializer/org.wso2.carbon.mediation.transport.handlers/src/main/java/org/wso2/carbon/mediation/transport/handlers/requestprocessors/swagger/format/SwaggerJsonProcessor.java

```diff
@@ -43,32 +44,45 @@
      * @param request              CarbonHttpRequest contains request information
      * @param response             CarbonHttpResponse which will be updated with response information
      * @param configurationContext axis2 configuration context
-     * @throws Exception if any exception occurred during definition generation
+     * @throws AxisFault  exception occurred during definition generation
      */
     public void process(CarbonHttpRequest request, CarbonHttpResponse response,
                         ConfigurationContext configurationContext) throws AxisFault {
-        API api = getAPIFromSynapseConfig(request);
 
-        if (api == null) {
+        API api = getAPIFromSynapseConfig(request);
+        String responseString = null;
+        if (api != null) {
+            responseString = processAPI(api, request);
+        } else if (request.getContextPath().contains("/" + SwaggerProcessorConstants.SERVICES_PREFIX)) {
```

Comment:
the format is 
host:port/services/blabla
host:port/t/tenantDomain/services/blabla
but this condition also pick
host:port/blabla/services

Does this get handled in SwaggerUtils.takeDataServiceSwagge